hail/src/test/scala/is/hail/io/compress/BGzipCodecSuite.scala

```diff
@@ -114,4 +115,78 @@ class BGzipCodecSuite extends SparkSuite {
     }
     p.check()
   }
+
+  @Test def testVirtualSeek() {
+    // real offsets of the start of each block, paired with the offset to the next block
+    val blockStarts = Array[(Long, Long)]((0, 14653), (69140, 82949), (133703, 146664), (181362, -1))
+    // magic number determined by indexing the target file
+    val maxBlockSize = 65280
+    // magic number determined by counting bytes from sample.vcf from uncompBlockStarts(-1) to the end of the file
+    val lastBlockLen = 55936
+    // offsets into the uncompressed file
+    val uncompBlockStarts = Array[Int](0, 326400, 652800, 913920)
+
+    val uncompPath = "src/test/resources/sample.vcf"
+    val compPath = "src/test/resources/sample.vcf.gz"
+
+    val uncompHPath = new hd.fs.Path(uncompPath)
+    val compHPath = new hd.fs.Path(compPath)
+
+    val fs = uncompHPath.getFileSystem(hadoopConf)
+    val compIS = fs.open(compHPath)
+    using (fs.open(uncompHPath)) { uncompIS => using (new BGzipInputStream(compIS)) { decompIS =>
+
+      for (((cOff, nOff), uOff) <- blockStarts.zip(uncompBlockStarts);
+           e <- Seq(0, 1024, maxBlockSize)) {
+        val decompData = new Array[Byte](100)
+        val uncompData = new Array[Byte](100)
+        val extra = if (cOff == blockStarts(blockStarts.size-1)._1 && e == maxBlockSize) {
```

Comment:
We don't normally put { } around single-statement values.  (Same comment in places below.)